pgpy/types.py

```diff
@@ -59,7 +59,7 @@ class Armorable(six.with_metaclass(abc.ABCMeta)):
     __armor_regex = re.compile(r"""# This capture group is optional because it will only be present in signed cleartext messages
                          (^-{5}BEGIN\ PGP\ SIGNED\ MESSAGE-{5}(?:\r?\n)
                           (Hash:\ (?P<hashes>[A-Za-z0-9\-,]+)(?:\r?\n){2})?
-                          (?P<cleartext>(.*\r?\n)+?)(?:\r?\n)?
+                          (?P<cleartext>(\r?\n|.)+?)(?:\r?\n)?
```

Comment:
That doesn't seem to match the oneline signature correctly. Try the armored blocks from #192, and my regression test from this PR.